mri/reconstruct/utils.py

```diff
@@ -207,19 +211,11 @@ def generate_operators(data, wavelet_name, samples, mu=1e-06, nb_scales=4,
             fourier_op=fourier_op)
 
     # Define the proximity dual/primal operator
-    prox_op = SparseThreshold(linear_op, mu, thresh_type="soft")
+    prox_op = SparseThreshold(Identity(), mu, thresh_type="soft")
```

Comment:
Why has this changed?

Reply:
Long story:
Generic cost has a calculate cost as follows:
 ```
        if self.analysis:
            cost = self.gradient_op.cost(x_new) + self.prox_op.cost(
                self.linear_op.op(x_new))
        else:
            cost = self.gradient_op.cost(x_new) + self.prox_op.cost(
                x_new)
```

Now you would notice that the cost call of proximity for synthesis formulation is with `x_new` which is the wavelet coefficient. If I place Sparse Threshold as my proximity with the same linear operator, we tend to find cost by trying to obtain the wavelet coefficients from `x_new` . This is not right. 
So the proximity is actually Sparsity of wavelet coefficients directly (hence the identity)

Reply:
But why don't we have the `prox_op` in the case of analysis have the `linear_op` ?
Most probably, `GenericCost` should not have to do this distinction.

Reply:
I dont think I understood you clearly. 
For the analysis formulation, x_new is actual image domain variable. 

So we could have a generic cost that just implements:
```python
cost = self.gradient_op.cost(x_new) + self.prox_op.cost(x_new)
```

and for `prox_op` have as follows:
```python
if self.analysis:
    prox_op = SparseThreshold(linear_op, mu, thresh_type="soft")
else:
    prox_op = SparseThreshold(Identity(), mu, thresh_type="soft")
```

This would essentially achieve the same thing. But yes, the `generic_cost` as a function would be more "generic"

Reply:
Yes it's the same thing in the end, but the setting of linear op or not is done at the right time.

Reply:
I agree, will resolve this